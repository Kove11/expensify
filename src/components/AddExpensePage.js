import React from 'react';
import {
  connect
}

  from 'react-redux';
import ExpenseForm from './ExpenseForm';
import {
  startAddExpense
}

  from '../actions/expenses';
export class AddExpensePage extends React.Component {
  onSubmit = (expense) => {
    // props.dispatch(addExpense(expense)); // old way to do it before using mapDispatchToProps
    this.props.startAddExpense(expense); // mapDispatchToProps basically wraps all function in a dispatch call
    this.props.history.push('/');
  }
    ;
  render() {
    return (<div> <div className="page-header"> <div className="content-container"> <h1 className="page-header__title">Add Expense</h1> </div> </div> <div className="content-container"> <ExpenseForm onSubmit={
      this.onSubmit
    }
    /> </div> </div>);
  }
}

const mapDispatchToProps = (dispatch) => ({
  startAddExpense: (expense) => dispatch(startAddExpense(expense))
}

);
export default connect(undefined, mapDispatchToProps)(AddExpensePage);