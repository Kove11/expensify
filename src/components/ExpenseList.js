import React from 'react';
import ExpenseListItem from './ExpenseListItem';
import {
    connect
}

    from 'react-redux';
import selectExpenses from '../selectors/expenses';
/// <ExpenseListItem {...item} />   spread operator passes all of the items from an object to the child
export const ExpenseList = (props) => (
    <div className="content-container">
        <div className="list-header">
            <div className="show-for-mobile">Expenses</div>
            <div className="show-for-desktop">Expense</div>
            <div className="show-for-desktop">Amount</div>
        </div>
        <div className="list-body">
            {
                props.expenses.length === 0 ? (
                    <div className="list-item list-item--message"><span >No expenses</span></div>)
                    : (
                        props.expenses.map((item) => (<ExpenseListItem key={
                            item.id
                        }
                            {
                            ...item
                            }
                        />)))
            }
        </div>
    </div>);
const mapStateToProps = (state) => {
    return {
        expenses: selectExpenses(state.expenses, state.filters)
    }
}

    ; //basically returns state into props
/// This is one of those wrapper components... two arguments, one to add state to props and one with component... returns a component that is conencted to redux with potentially a subset of the state
export default connect(mapStateToProps)(ExpenseList);