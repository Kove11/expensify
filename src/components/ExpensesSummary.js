import React from 'react';
import selectExpenses from '../selectors/expenses';
import selectExpensestotal from '../selectors/expenses-total';
import {connect} from 'react-redux';
import numeral from 'numeral';

export const ExpensesSummary = ({expenseCount, expensesTotal}) => (

     <h1>Viewing {expenseCount} {expenseCount === 1 ? "expense" : "expenses"} totalling {numeral(expensesTotal/100).format('$0,0.00')}</h1>

);
const mapStateToProps = (state) => { return { expenseCount: selectExpenses(state.expenses,state.filters).length, expensesTotal: selectExpensestotal(selectExpenses(state.expenses,state.filters))  }};

export default connect(mapStateToProps)(ExpensesSummary);