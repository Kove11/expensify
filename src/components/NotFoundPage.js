import React from 'react';
import {Link} from 'react-router-dom';
const NotFoundPage = () => (
    <div>
      Page Not Found - You probably shouldn't be here.<br/>
      <Link to="/">Back to safety</Link>
    </div>
  );

export default NotFoundPage;