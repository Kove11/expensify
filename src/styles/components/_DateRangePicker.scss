.DateRangePickerInput {
    @media (max-width: $desktop_breakpoint) {
        //When it's mobile size these styles will apply
        width: 100%;
        display: flex;
        justify-content: space-between;
        align-items: center;
    }
}