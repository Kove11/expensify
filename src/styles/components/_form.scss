.form {
    display: flex;
    flex-direction: column;
    >* {
        // this selects all direct children of items with form class
        margin-bottom: $m-size
    }
}

.form__error {
    margin: 0 0 $m-size 0;
    font-style: italic;
}