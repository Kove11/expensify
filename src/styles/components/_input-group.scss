.input-group {
    display: flex;
    margin-bottom: $m-size;
    flex-direction: column;
    @media(min-width: $desktop_breakpoint) {
        flex-direction: row;
        margin-bottom: $l-size;
    }
}

.input-group__item {
    margin-bottom: $s-size;
    @media(min-width: $desktop_breakpoint) {
        margin:0 $s-size 0 0;
    }
}