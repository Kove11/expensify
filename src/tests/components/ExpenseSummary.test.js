import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';


test('should render expense summary with 1 item', ()=>{
    const wrapper = shallow(<ExpensesSummary expensesTotal={9434} expenseCount={1}/>);
    expect(wrapper).toMatchSnapshot();
});

test('should render expense summary with 2 items', ()=>{
    const wrapper = shallow(<ExpensesSummary  expenseCount={2} expensesTotal={9434}/>);
    expect(wrapper).toMatchSnapshot();
});