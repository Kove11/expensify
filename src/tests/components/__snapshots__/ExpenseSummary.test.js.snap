// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render expense summary with 1 item 1`] = `
<h1>
  Viewing 
  1
   
  expense
   totalling 
  $94.34
</h1>
`;

exports[`should render expense summary with 2 items 1`] = `
<h1>
  Viewing 
  2
   
  expenses
   totalling 
  $94.34
</h1>
`;
